ttnn/cpp/ttnn/operations/pool/downsample/device/downsample_op.cpp

```diff
@@ -30,27 +30,17 @@ void Downsample::validate(const std::vector<Tensor>& input_tensors) const {
         input_tensor_a.memory_config().memory_layout);
 }
 
-std::vector<tt::tt_metal::LegacyShape> Downsample::compute_output_shapes(
-    const std::vector<Tensor>& input_tensors) const {
-    const auto& input_tensor_a = input_tensors.at(0);
-    TT_ASSERT(input_tensor_a.get_legacy_shape()[0] == 1 && input_tensor_a.get_legacy_shape()[1] == 1);
-    uint32_t input_height = input_tensor_a.get_legacy_shape()[2];
+std::vector<TensorSpec> Downsample::compute_output_specs(const std::vector<Tensor>& input_tensors) const {
+    const auto& input_tensor = input_tensors.at(0);
+    TT_ASSERT(input_tensor.get_legacy_shape()[0] == 1 && input_tensor.get_legacy_shape()[1] == 1);
+    uint32_t input_height = input_tensor.get_legacy_shape()[2];
     auto [img_batch_size, img_height, img_width, img_stride_h, img_stride_w] = this->downsample_params;
     TT_ASSERT(input_height >= img_batch_size * img_height * img_width);
-    uint32_t output_height_unpadded = img_batch_size * ceil((double)img_height / (double)img_stride_h) *
-                                      ceil((double)img_width / (double)img_stride_w);
-    uint32_t output_height = tt::round_up(output_height_unpadded, TILE_HEIGHT);
-    uint32_t output_width = input_tensor_a.get_legacy_shape()[3];
-    auto output_padding =
```

Comment:
Seems like this op was trying to encode padding info in it's output tensor, how is handled now?

Reply:
This computation is not needed anymore. It is now properly handled by tensor layout.

Reply:
Ok I'll trust you on that